### PR TITLE
Added a component template

### PR DESCRIPTION
**Pull request by GreenAsJade**
Fixes "darn, now how do I do this again?"

## Proposed Changes

Add a 'templates' directory and wack an example component in it with various things that we always need.

### CODE REVIEW

File: src/templates/component/TemplateComponent/index.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (C) 2012-2017  Online-Go.com
```
Comment (anoek) [inline src/templates/component/TemplateComponent/index.ts:2]: geeze where did you dig that year from? Guess I missed the source for that a few times :sweat_smile:

Comment (GreenAsJade) [reply src/templates/component/TemplateComponent/index.ts:2]: LOL yeah ... if you grep it you'll find plenty of hits, it turns out!  :o

File: src/templates/component/TemplateComponent/TemplateComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+type TemplateComponentProps = {
+    render_twice?: boolean;
+};
+
+export function TemplateComponent(props: TemplateComponentProps): JSX.Element {
+    const [username, setUsername] = React.useState("");
+
+    const hangs_around = React.useRef("forever");
+
+    const user = data.get("config.user"); // user.anonymous will be set if they aren't logged in
```
Comment (anoek) [inline src/templates/component/TemplateComponent/TemplateComponent.tsx:33]: better to use `const user = useUser();` these days, comes from `import { useUser } from "hooks";`

File: src/templates/component/TemplateComponent/index.ts (hunk 1 above)
Comment (GreenAsJade) [reply src/templates/component/TemplateComponent/index.ts]: ... yet another reason for having a template that is correct :D

File: src/templates/component/TemplateComponent/TemplateComponent.tsx (hunk 2 above)
Comment (GreenAsJade) [reply src/templates/component/TemplateComponent/TemplateComponent.tsx:33]: 😮    ... and another reason to have learnings in a template!

Comment (anoek) [reply src/templates/component/TemplateComponent/TemplateComponent.tsx:33]: (useUser will automatically refresh when the user object is updated, as in when the username/icon/ui_class/rank changes kinda stuff)

Comment (anoek) [reply src/templates/component/TemplateComponent/TemplateComponent.tsx:33]: Yeah it's pretty new, figured if we're making a template though, that should be in there :)

Comment (GreenAsJade) [reply src/templates/component/TemplateComponent/TemplateComponent.tsx]: Template's been there <24 hours, already winning :D  :D

**Pull request merged**